### PR TITLE
feature: stem installs to workflow name not `rose-stem`

### LINKED ISSUE #177: Rose stem installs to `~/cylc-run/rose-stem`
**Describe the bug**

When `$PWD` is not a subversion repo `rose stem --source=X=$PWD` is unable to get a name for the workflow install directory and installs the workflow in `~/cylc-run/rose-stem`. Ideally the basename of the source directory should be used.

**Expected behavior**
```diff
rose stem --source=X=/home/h02/tpilling/cylc-src/example
...
- INSTALLED example/run1 from /net/home/h02/tpilling/cylc-src/example/rose-stem
+ INSTALLED rose-stem/run3 from /net/home/h02/tpilling/cylc-src/example/rose-stem
```

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

### PR DESCRIPTION
**Pull request by wxtim**
These changes close #177 

## Requirements check-list

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to `setup.cfg`.
- [x] Appropriate tests are included (unit and/or functional).
- [x] Appropriate change log entry included.
- [x] No documentation update required.

### CODE REVIEW

**Comment by oliver-sanders** [comment]
Pretty sure this is a bug right (otherwise it goes in 1.2.0), swapped "enhancement" for "bug" to match the issue.

**Comment by wxtim** [comment]
> Pretty sure this is a bug right (otherwise it goes in 1.2.0), swapped "enhancement" for "bug" to match the issue.

It's the same behaviour as Rose 2019.

**Review by datamel — APPROVED** [review]
I have run the tests on master and they fail, they pass locally for me on this branch. I have also read the code and manually tested this change, using the documented command in the issue. 
Testing this out, I had a play around and rose stem is not coping with source with a `~`, expanding it myself is working. Would this be a desirable feature for our users? If so I can open another issue.

File: cylc/rose/stem.py [hunk 1]
```diff
@@ -369,7 +369,7 @@ def _generate_name(self):
                 basedir = os.path.abspath(self.opts.source)
             else:
                 basedir = os.getcwd()
-        name = os.path.basename(basedir)
+        name = os.path.basename(os.path.dirname(basedir))
```
Comment (dpmatthews) [inline cylc/rose/stem.py:372]: This is in the wrong place - it will change the name in all cases rather than just the case we're trying to fix

**Review by dpmatthews — CHANGES_REQUESTED** [review]

File: tests/unit/test_rose_stem_units.py [hunk 2]
```diff
@@ -54,3 +55,48 @@ def test_get_source_opt_from_args(tmp_path, monkeypatch, args, expect):
...
+        class MonkeyStemRunner(StemRunner):
+            def _ascertain_project(self, x):
+                return ['foo', '/foo/bar']
+        return MonkeyStemRunner(SimpleNamespace, reporter=print)
+
+    @staticmethod
+    def source_set(monkeypatch):
+        """It gets the name from the source opt."""
+        opts = SimpleNamespace()
+        opts.source = 'bar/rose-stem'
```
Comment (dpmatthews) [inline tests/unit/test_rose_stem_units.py:80]: `/foo/bar/rose-stem` might be a slightly better test?

Comment (wxtim) [reply tests/unit/test_rose_stem_units.py:80]:
```suggestion
        opts.source = '/foo/bar/rose-stem'
```

**Review by datamel — APPROVED** [review]
LGTM. I have checked this out (rebased to get latest `setup.cfg`), run tests locally and used the example in the issue to test it is working as expected.

File: cylc/rose/stem.py [hunk 3]
```diff
@@ -17,24 +17,28 @@
...
-This command acts as a wrapper to "cylc install" by defining a number of
-additional Jinja2 variables.
+Install a rose-stem suite using cylc install.
 
-The path to the workflow source directory can be specified using the
-"path" argument. Otherwise it must be provided in a directory named
-"rose-stem" in the first source tree.
+To run a rose-stem suite use "cylc play".
+
+Install a suitable suite with a specified set of source tree(s).
```
Comment (wxtim) [inline cylc/rose/stem.py:26]:
```suggestion
Install a suitable suite with a specified set of source tree(s).

To run a rose-stem suite use "cylc play".
```

**Review by MetRonnie — APPROVED** [review]

**Pull request merged**